frontend/src/concepts/pipelines/topology/PipelineTopology.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { PipelineNodeModel, VisualizationProvider } from '@patternfly/react-topology';
+import { Bullseye, Spinner } from '@patternfly/react-core';
+import useTopologyController from '~/concepts/pipelines/topology/useTopologyController';
+import PipelineVisualizationSurface from '~/concepts/pipelines/topology/PipelineVisualizationSurface';
+
+type PipelineTopologyProps = {
+  nodes: PipelineNodeModel[];
+};
+
+const PipelineTopology: React.FC<PipelineTopologyProps> = ({ nodes }) => {
+  const controller = useTopologyController('g1');
+
+  if (!controller) {
```

Comment:
will there ever be a failure case? B/c right now it would just spin right?

Reply:
The hook is weird -- it does the effort behind the useEffect to set a controller -- it should "never" fail. Not sure even how we would want to handle that... I suspect if it fails, it throws an exception. 

The flow of the hook is:
- Initial setup -- no controller (`null`) => Show spinner
- Hook triggers -- sets controller => Show graph

ErrorBoundaries might be a course of action here. But not part of this PR, it should be done in `main` and brought in. We have no error boundaries and we should at least do one for the content pane.